docs/docsite/_ext/ansible_redirects.py

```diff
@@ -0,0 +1,27 @@
+"""Sphinx extension for generating redirects in Ansible package docs."""
+
+import os
+from pathlib import Path
+
+import yaml
+from sphinx.application import Sphinx
+
+DOCSITE_ROOT_DIR = Path(__file__).parents[1].resolve()
+ANSIBLE_REDIRECTS_CONFIG_PATH = DOCSITE_ROOT_DIR / "ansible_redirects.yaml"
+REDIRECT_KEYS = yaml.safe_load(
+    ANSIBLE_REDIRECTS_CONFIG_PATH.read_text(encoding="utf-8")
+)
+ENABLE_REDIRECTS = os.environ.get("ENABLE_REDIRECTS", "false").lower() in {"1", "true"}
+
+
+def setup(app: Sphinx):
```

Comment:
an annotation is missing for the return value

Reply:
Thanks for spotting. I had this in draft as I knew I needed to go back and add annotations, docstrings and some more comments. Guess I should always just do that stuff straightaway instead of in retrospect.